pandas/core/index.py

```diff
@@ -2580,7 +2580,10 @@ def drop(self, labels, errors='raise'):
     @Appender(_shared_docs['drop_duplicates'] % _index_doc_kwargs)
     def drop_duplicates(self, take_last=False):
         result = super(Index, self).drop_duplicates(take_last=take_last)
-        return self._constructor(result)
+        if self.name is not None:
```

Comment:
I would even be happy if you just explicitly check `isinstance(self, MultiIndex)`.

I agree that your alternative solution would be preferable. That's part of a much bigger issue: https://github.com/pydata/pandas/issues/3268